### PR TITLE
fix(aws)!: PK for `aws_eventbridge_event_bus_rules`

### PR DESCRIPTION
**Pull request by bbernays**

### CODE REVIEW

File: plugins/source/aws/resources/services/eventbridge/event_bus_rules.go [hunk 1]
```diff
@@ -35,6 +35,13 @@ func EventBusRules() *schema.Table {
 				Type:     schema.TypeJSON,
 				Resolver: resolveEventbridgeEventBusRuleTags,
 			},
+			{
+				Name: "arn",
```
Comment (candiduslynx) [inline plugins/source/aws/resources/services/eventbridge/event_bus_rules.go:39]: use [transformers.WithPrimaryKeys](https://pkg.go.dev/github.com/cloudquery/plugin-sdk/transformers#WithPrimaryKeys) option and not the extra column.
Additionally, should drop the `event_bus_arn` column as duplicate

Comment (bbernays) [reply plugins/source/aws/resources/services/eventbridge/event_bus_rules.go:39]: `event_bus_arn` is the parent arn...

Comment (candiduslynx) [reply plugins/source/aws/resources/services/eventbridge/event_bus_rules.go:39]: Ah, sorry, then dismissing that part.
The transformation & ad-hoc columns still stand, though

**Review by candiduslynx — APPROVED** [review]

File: plugins/source/aws/resources/services/eventbridge/event_bus_rules.go [hunk 2]
```diff
@@ -35,6 +35,13 @@ func EventBusRules() *schema.Table {
...
+				Type: schema.TypeString,
+				CreationOptions: schema.ColumnCreationOptions{
+					PrimaryKey: true,
+				},
```
Comment (hermanschaaf) [inline plugins/source/aws/resources/services/eventbridge/event_bus_rules.go:43]: I think we'll need to specify the resolver here

**Review by hermanschaaf — CHANGES_REQUESTED** [review]
Looks like there's a missing path resolver, but let me know if I misunderstood somethig

File: plugins/source/aws/resources/services/eventbridge/event_bus_rules.go (hunk 2 above)
Comment (bbernays) [reply plugins/source/aws/resources/services/eventbridge/event_bus_rules.go:43]: If I needed the resolver, wouldn't the unit tests fail?

Comment (candiduslynx) [reply plugins/source/aws/resources/services/eventbridge/event_bus_rules.go:43]: [This](https://github.com/cloudquery/plugin-sdk/blob/main/transformers/struct.go#L258-L260) should skip adding from struct with default transformer.
So the following [code](https://github.com/cloudquery/plugin-sdk/blob/9f1e373b516be958f0594e84fbbbcd43951f14ad/plugins/source/scheduler.go#L117-L133) works instead (col resolver = `nil`).

I think it's better to:
1. Add resolver explicitly for explicitly defined columns
2. Add a check (in transform? or during plugin validate phase? but better in a static place) that all columns have resolvers defined.
3. Remove fallback for case with `nil` column resolver.

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**